In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
trade_history = pd.read_csv('data/processed.csv')

## Week Based Analysis
Day of week based comparison of trade count, pnl mean and sum

In [3]:
# overall totals without classification
overall = trade_history.groupby(
    'day_week'
).agg({
    'Trade ID': 'count',
    'Closed PnL': ['sum', 'mean']
})

overall

Trade ID    Closed PnL           
             count           sum       mean
day_week                                   
Friday       31541  1.875712e+06  59.469006
Monday       31859  1.300380e+06  40.816708
Saturday     22826  1.187535e+06  52.025555
Sunday       22044  1.401941e+06  63.597408
Thursday     35239  1.922707e+06  54.561897
Tuesday      31545  1.959181e+06  62.107492
Wednesday    36164  6.070314e+05  16.785517

## Key Insights: Weekly Trading Patterns

- `Wednesday` had the highest trading activity:
  - **36,164 trades**
  - But generated the lowest average pnl:
    - **$16.79 avg pnl**

- `Sunday` was the most profitable trading day:
  - **$63.60 avg pnl**
  - Total pnl:
    - **$1.40M**

- `Tuesday` generated the highest total pnl:
  - **$1.96M**
  - Avg pnl:
    - **$62.11**

- `Thursday` and `Friday` also showed strong profitability:
  - Thursday:
    - **$1.92M total pnl**
  - Friday:
    - **$59.47 avg pnl**

- `Monday` showed relatively weak trading performance:
  - Avg pnl:
    - **$40.82**

- Higher trade activity did not correspond to higher profitability:
  - Example:
    - Wednesday → **36.1K trades**, only **$16.79 avg pnl**
    - Sunday → **22K trades**, but **$63.60 avg pnl**

- Weekend trading (`Saturday–Sunday`) showed stronger average profitability compared to midweek trading periods.

Day of week vs Sentiment based comparison of trade count, pnl mean and sum

In [4]:
# with classification
classified = trade_history.groupby(
    ['day_week', 'classification']
).agg({
    'Trade ID': 'count',
    'Closed PnL': ['sum', 'mean']
})

classified

Trade ID     Closed PnL            
                            count            sum        mean
day_week  classification                                    
Friday    Extreme Fear       6918  260741.560751   37.690309
          Extreme Greed      6222  702341.899692  112.880408
          Fear               6941   79807.173904   11.497936
          Greed              8388  547197.396718   65.235741
          Neutral            3072  285623.887807   92.976526
Monday    Extreme Fear       4229  -41981.334238   -9.927012
          Extreme Greed      4313  319589.574472   74.099136
          Fear               6263  686353.377843  109.588596
          Greed              6231  114841.765554   18.430712
          Neutral           10823  221576.125382   20.472709
Saturday  Extreme Fear       1767  269595.683965  152.572543
          Extreme Greed      5682  267594.794745   47.095177
          Fear               8304  264041.691688   31.796928
          Greed              4994  195114.464236   39.069777
          Neutral            2079  191188.673423   91.961844
Sunday    Extreme Fear         42   -1104.856765  -26.306113
          Extreme Greed      5027  280271.397981   55.753212
          Fear               6936  466684.377043   67.284368
          Greed              5176  525255.017955  101.478945
          Neutral            4863  130835.319541   26.904240
Thursday  Extreme Fear       7268  188630.131648   25.953513
          Extreme Greed     10133  757949.404623   74.800099
          Fear               7445  690694.341999   92.772914
          Greed              5227  188768.190365   36.114060
          Neutral            5166   96664.629061   18.711697
Tuesday   Extreme Greed      4574  300244.050278   65.641463
          Fear              13141  698274.311638   53.137076
          Greed              6396  602247.661582   94.160047
          Neutral            7434  358414.801772   48.212914
Wednesday Extreme Fear       1176   63229.063129   53.766210
          Extreme Greed      4041   87180.188882   21.573915
          Fear              12807  471300.167527   36.800200
          Greed             13891  -23295.223459   -1.677001
          Neutral            4249    8617.238564    2.028063

# Key Insights: Weekly Sentiment-Based Trading Patterns

## Friday
- `Extreme Greed` generated the highest avg pnl:
  - **$112.88 avg pnl**
- `Neutral` sentiment also performed strongly:
  - **$92.98 avg pnl**

## Monday
- `Fear` produced the strongest profitability:
  - **$109.59 avg pnl**
- `Extreme Fear` was loss-making:
  - **-$9.93 avg pnl**

## Saturday
- `Extreme Fear` showed the highest profitability across all weekday-sentiment combinations:
  - **$152.57 avg pnl**
- `Neutral` also remained highly profitable:
  - **$91.96 avg pnl**

## Sunday
- `Greed` performed exceptionally well:
  - **$101.48 avg pnl**
- `Extreme Fear` remained highly unprofitable:
  - **-$26.31 avg pnl**

## Thursday
- `Fear` and `Extreme Greed` dominated profitability:
  - Fear → **$92.77 avg pnl**
  - Extreme Greed → **$74.80 avg pnl**

## Tuesday
- `Greed` achieved strong profitability:
  - **$94.16 avg pnl**
- `Fear` also showed high activity:
  - **13,141 trades**

## Wednesday
- `Greed` became loss-making:
  - **-$1.68 avg pnl**
- `Neutral` sentiment showed extremely weak profitability:
  - **$2.03 avg pnl**

- `Fear` sentiment consistently maintained high trade participation throughout the week.
- Weekend trading (`Saturday–Sunday`) showed significantly higher profitability under extreme sentiment regimes compared to midweek periods.
- `Extreme Fear` produced highly volatile outcomes:
  - Best → Saturday (**$152.57 avg pnl**)
  - Worst → Sunday (**-$26.31 avg pnl**)